clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -547,7 +547,7 @@ public int read(ByteBuffer dst) throws IOException {
                 } else if (unwrapResult.getStatus() == Status.BUFFER_OVERFLOW) {
                     int currentApplicationBufferSize = applicationBufferSize();
                     appReadBuffer = Utils.ensureCapacity(appReadBuffer, currentApplicationBufferSize);
-                    if (appReadBuffer.position() >= currentApplicationBufferSize) {
+                    if (appReadBuffer.position() > currentApplicationBufferSize) {
                         throw new IllegalStateException("Buffer overflow when available data size (" + appReadBuffer.position() +
                                                         ") >= application buffer size (" + currentApplicationBufferSize + ")");
```

Comment:
Need to change ">=" to ">" in the exception statement here.

Reply:
thanks @ganeshmv5 will do.